lib/lotus/validations/attribute_definer.rb

```diff
@@ -0,0 +1,273 @@
+module Lotus
+  module Validations
+    # Define attributes and their validations together
+    module AttributeDefiner
+      def self.included(base)
+        base.extend ClassMethods
+      end
+
+      module ClassMethods
+        # Define an attribute
+        #
+        # @param name [#to_sym] the name of the attribute
```

Comment:
This is my first time see this documenting convention `#to_sym`, does that mean we could parse in object that respond to `to_sym`?